packages/puppeteer-core/src/api/Page.ts

```diff
@@ -1724,6 +1732,38 @@ export abstract class Page extends EventEmitter<PageEvents> {
     options?: {timeout?: number}
   ): Promise<HTTPResponse>;
 
+  /**
+   * @internal
+   */
+  async _waitForHTTP<HTTP extends {url(): string}>(
+    networkManager: EventEmitter<NetworkManagerEvents>,
```

Comment:
do we need to type-safe events here? let's define am interface we need? e.g., events with something that emits HTTP